astropy/units/tests/test_quantity.py

```diff
@@ -1055,12 +1055,31 @@ def test_quantity_pickelability():
     assert q1.unit == q2.unit
 
 
-def test_quantity_from_string():
-    with pytest.raises(TypeError):
-        q = u.Quantity(u.m * "5")
-        # the reverse should also fail once #1408 is in
+def test_quantity_initialisation_from_unit():
+    q = u.Quantity(u.m)
+    assert q.unit == u.m
+    assert q.value == 1.
+    assert u.Unit(q) == u.m
+    q = u.Quantity('m')
```

Comment:
I'm less ambivalent. My understanding is that a `Quantity` is the combination of a unit and a scalar. `m` is a unit, thus `Quantity('m')` is not a quantity. This should be undefined. This is closer to a mistake - I do not suggest assuming the value of "1" here. Why not zero? That seems equally valid. It’s ambiguous.